### PR TITLE
Feature/advanced filter fab

### PR DESCRIPTION
**Pull request by esotrope**
Before click:
screen shot 2017-08-30 at 4 14 09 pm

After click:
screen shot 2017-08-30 at 4 14 23 pm

### CODE REVIEW

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts [hunk 1]
```diff
@@ -227,6 +227,13 @@ export class AssessmentsComponent implements OnInit {
     this.onExportItemsByPointsEarned.emit(exportRequest);
   }
 
+  openAndScrollToAdvancedFilters() {
+    this.showAdvancedFilters = true;
+    setTimeout(() => {
+      document.getElementById('results-adv-filters').scrollIntoView();
+    }, 0);
+  }
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: open to suggestions on a good way to not duplicate this code on both assessment result pages (student / group+school)

File: webapp/src/main/webapp/src/styles.less [hunk 2]
```diff
@@ -114,6 +114,13 @@ footer {
   margin-bottom: @line-height-computed / 2;
 }
 
+.fab {
+  position: fixed;
+  right: @grid-gutter-width / 2;
+  bottom: @line-height-computed / 2;
+  z-index: 2000;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/styles.less]: @aaronsleeper had to add zindex to make the button appear over the assessment result tables.

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts (hunk 1 above)
Comment (bgsavage) [reply webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: I think the angular way is to create a `@ViewChild` and then tag the element with that `#name` to get at the element.

Comment (bgsavage) [reply webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: re: dup code - could this button live in the adv filters component?

File: webapp/src/main/webapp/src/assets/i18n/en.json [hunk 3]
```diff
@@ -234,7 +234,8 @@
             "limited-english-proficiency-info": "Students who use a language other than English as their primary language and who have been assessed as not possessing the English language skills necessary to succeed in the regular instructional program.",
             "ethnicity": "Race/Ethnicity",
             "ethnicity-info": "Student race and ethnicity"
-          }
+          },
+          "fab": "Filter Results"
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/assets/i18n/en.json]: nit: abbreviations - what's a fab? :)

File: webapp/src/main/webapp/src/styles.less (hunk 2 above)
Comment (bgsavage) [reply webapp/src/main/webapp/src/styles.less]: nit: may want to consider defining variables of known z-indexes so it is known why 2000 was chosen.

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts (hunk 1 above)
Comment (aaronsleeper) [reply webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: the screenshots above look right to me. thanks :)

Comment (esotrope) [reply webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: on dup code - could do that but would have to pass in callbacks to advance filters like: ```(fabClick)=this.showAdvanceFilters=true``` not sure if that is better or not. Also, conceptually I feel like the button should live beside advanced filters and not in the component, but that's my feel.

File: webapp/src/main/webapp/src/styles.less (hunk 2 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/styles.less]: maybe ```@z-index-greater-than-tables``` ?

File: webapp/src/main/webapp/src/assets/i18n/en.json (hunk 3 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/assets/i18n/en.json]: You're fab Brandon : j

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: Realized the show/hide will show/hide the fab if we put it in the advanced filter component. I think i'll leave it as is for now.
```html
<!-- Advanced Filters -->
   <div id="group-results-adv-filter-options" [hidden]="!showAdvancedFilters || !filterDisplayOptions.expanded" class="well gray-lighter">
     <adv-filters [filterBy]="clientFilterBy" [filterOptions]="filterOptions"></adv-filters>
   </div>
```

**Pull request merged**